packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1179,10 +1193,11 @@ abstract class ResidentRunner {
   Future<void> preExit() async {
     // If _dillOutputPath is null, we created a temporary directory for the dill.
     if (_dillOutputPath == null && artifactDirectory.existsSync()) {
-      final File outputDill = artifactDirectory.childFile('app.dill');
+      final File outputDill = globals.fs.file(dillOutputPath);
```

Comment:
Why did this line change? `_dillOutputPath` could be null here

Reply:
Yep, but here's how `dillOutputPath` defined
```dart
String get dillOutputPath => _dillOutputPath ?? globals.fs.path.join(artifactDirectory.path, 'app.dill');
```

There were only 2 places where `app.diil` was hard-coded and I thought it was a good idea to reduce this to 1

Reply:
Ahh makes sense, thanks!